network/endpoint_linux.go

```diff
@@ -94,18 +94,24 @@ func (nw *network) newEndpointImpl(
 	} else {
 		// Create a veth pair.
 		logger.Info("Generate veth name based on endpoint id")
-		hostIfName = fmt.Sprintf("%s%s", hostVEthInterfacePrefix, defaultEpInfo.Id[:7])
-		contIfName = fmt.Sprintf("%s%s-2", hostVEthInterfacePrefix, defaultEpInfo.Id[:7])
+		hostIfName = fmt.Sprintf("%s%s", hostVEthInterfacePrefix, defaultEpInfo.EndpointID[:7])
+		contIfName = fmt.Sprintf("%s%s-2", hostVEthInterfacePrefix, defaultEpInfo.EndpointID[:7])
+	}
+
+	nicName := epInfo.IfName
+	// infra nic nicname will look like eth0, but delegated/secondary nics will look like "eth1 eth2 eth3"
+	if epInfo.NICType != cns.InfraNIC {
+		nicName = epInfo.MasterIfName
 	}
```

Comment:
cant we set epInfo.Ifname in cni/network when creating epInfo based on nic type? we can avoid creating another field NicName